crates/wysiwyg/src/composer_model.rs

```diff
@@ -354,6 +436,61 @@ where
             panic!("Trying to bold a non-text node")
         }
     }
+
+    fn add_list_item(&mut self, handle: DomHandle, location: usize) {
+        let list_node = self.state.dom.lookup_node_mut(handle);
+        if let DomNode::Container(list) = list_node {
+            list.append(DomNode::new_list_item(
+                "li".to_html(),
+                vec![DomNode::Text(TextNode::from("\u{200B}".to_html()))],
+            ));
+            self.state.start = Location::from(location + 1);
+            self.state.end = Location::from(location + 1);
+        } else {
+            panic!("Handle doesn't match a list container node")
+        }
+    }
+
+    fn remove_list_item(
+        &mut self,
+        handle: DomHandle,
+        location: usize,
+        list_item_index: usize,
+    ) {
+        let list_node = self.state.dom.lookup_node_mut(handle.clone());
+        if let DomNode::Container(list) = list_node {
+            if list.children().len() == 1 {
+                let parent_handle = handle.parent_handle();
+                let parent_node = self.state.dom.lookup_node_mut(parent_handle);
+                if let DomNode::Container(parent) = parent_node {
+                    parent.remove(handle.index_in_parent());
+                    if parent.children().len() == 0 {
+                        parent.append(DomNode::Text(TextNode::from(
+                            "".to_html(),
+                        )));
+                    }
+                    self.state.start = Location::from(location);
+                    self.state.end = Location::from(location);
+                } else {
+                    panic!("List has no parent container")
```

Comment:
Longer term, we should handle `li` not inside `ol` or `ul` - this could happen in HTML, even though we won't generate it. Probably fine for it to be a TODO for now, but we should think about it.